scipy/optimize/zeros.py

```diff
@@ -819,3 +838,480 @@ def brenth(f, a, b, args=(),
         raise ValueError("rtol too small (%g < %g)" % (rtol, _rtol))
     r = _zeros._brenth(f, a, b, xtol, rtol, maxiter, args, full_output, disp)
     return results_c(full_output, r)
+
+
+################################
+# TOMS "Algorithm 748: Enclosing Zeros of Continuous Functions", by
+#  Alefeld, G. E. and Potra, F. A. and Shi, Yixun,
+#  See [1]
+
+
+def _within_tolerance(x, y, rtol, atol):
```

Comment:
replace with `np.allclose`? seems to do the same

Reply:
`np.isclose` has a lot of overhead related to its support for arrays, so I've taken to using a local implementation of `within_tol` which doesn't have some of that overhead.  In this particular case, `_within_tolerance` is only called once per iteration, so it may not matter.

Reply:
fair enough - fine to leave as is. may a comment that this should be equivalent to `np.isclose` but leaves out the unnecessary checks would be useful, to prevent someone in the future from "cleaning this up".